test/boost/mutation_reader_test.cc

```diff
@@ -3848,7 +3848,7 @@ static future<> do_test_clustering_order_merger_sstable_set(bool reversed) {
     auto pr = dht::partition_range::make_singular(dht::ring_position(g._pk));
     auto make_tested = [&env, query_schema, pk = g._pk, &pr, &query_slice, reversed]
             (const time_series_sstable_set& sst_set,
-                const std::unordered_set<int64_t>& included_gens, streamed_mutation::forwarding fwd) {
+                const std::unordered_set<sstables::generation_type::int_t>& included_gens, streamed_mutation::forwarding fwd) {
```

Comment:
Why not use generation_type, here and elsewhere?

Reply:
it is converted to that later on in the series, once possible

Reply:
Also generation_type(int_t) ctor is defined as explicit, on purpose to catch inadvertent type conversion.
If we make this change early, then we need to allow implicit conversion and maybe by the end of the series make it explicit again after making all intended conversions explicit.